workflow-controller/controller-service/src/main/java/io/hyscale/controller/Converters/Converter.java

```diff
@@ -0,0 +1,134 @@
+
+    @Autowired
+    private SchemaValidator schemaValidator;
+
+    private static final Logger logger = LoggerFactory.getLogger(Converter.class);
+
+    public abstract String getFilePattern();
+
+    public abstract HyscaleSpecType getReferenceSchemaType();
+
+    public abstract ServiceSpecActivity getWarnMessage();
+
+    public abstract boolean validateData(File inputFile) throws HyscaleException;
+
+    /**
+     * Performs command line input validation and assigns to the respective field.
+     *
+     * <ul>
+     *  <li>if input found empty throws Empty file path Hyscale Exception.</li>
+     *  <li>if input file does not exits throws File not found Hyscale Exception.</li>
+     *  <li>if input is not file throws Cannot process file Hyscale Exception.</li>
+     *  <li>if input file is empty throws Empty file Hyscale Exception.</li>
+     *  <li>if input does not satisfy the respective reference schema throws Invalid Format Hyscale Exception.</li>
+     *  </ul>
+     * @param inputFilePath
+     * @return assigns the file to field if valid.
+     * @throws Exception contains wrapped HyscaleExcpetion for Handling.
+     */
+    @Override
+    public List<File> convert(String inputFilePath) throws Exception {
+        ProcessingReport processingReport;
+        List<File> inputFiles = new ArrayList<File>();
+
+        if (StringUtils.isBlank(inputFilePath)) {
+            WorkflowLogger.error(ControllerActivity.EMPTY_FILE_PATH);
+            throw new HyscaleException(CommonErrorCode.EMPTY_FILE_PATH, ToolConstants.INVALID_SPEC_ERROR_CODE);
+        }
+
+        File specfile = new File(inputFilePath);
+        if (!specfile.exists()) {
+            WorkflowLogger.error(ControllerActivity.CANNOT_FIND_FILE, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.FILE_NOT_FOUND,ToolConstants.INVALID_SPEC_ERROR_CODE,inputFilePath);
+        }
+        if (specfile.isDirectory()) {
+            WorkflowLogger.error(ControllerActivity.DIRECTORY_INPUT_FOUND, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.FOUND_DIRECTORY_INSTEAD_OF_FILE,ToolConstants.INVALID_SPEC_ERROR_CODE,inputFilePath);
+        }
+        if (!specfile.isFile()) {
+            WorkflowLogger.error(ControllerActivity.INVALID_FILE_INPUT, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.INVALID_FILE_INPUT,ToolConstants.INVALID_SPEC_ERROR_CODE);
+        }
+        String fileName = specfile.getName();
+        if (!fileName.matches(getFilePattern())) {
+            WorkflowLogger.warn(getWarnMessage(),fileName);
+            logger.warn(getWarnMessage().getActivityMessage(), fileName);
+        }
+        if (specfile.length() == 0) {
+            WorkflowLogger.error(ControllerActivity.EMPTY_FILE_FOUND, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.EMPTY_FILE_FOUND,ToolConstants.INVALID_SPEC_ERROR_CODE, inputFilePath);
+        }
```

Comment:
We can extract these to a private method say validate file??

Reply:
will discuss